Add non-minified theme.css

sivel opened the pull request:
The purpose of this pull request is to add a non-minified version of theme.css so that it will be easier to update in the future.

A new `staticmin` make target has been created and is in use by docs, viewdocs, and htmldocs.  clean has also been created to remove the created min.css file.

sivel committed:
Add beautified theme.css

* Update theme.conf to use theme.min.css
* Add "beautified" (un-minified) theme.css

mpdehaan commented:
So I see the rename of the theme.min.css though do we also want to check in a unminified copy?

I'm thinking maybe the 'git add' was left out on this file.

Though I see this is normally generated with sass or compass or something trendy I think having a non-minified but non-sass/other copy in there is probably not a bad thing, but that the makefile target should probably call the minifier to avoid checking in the build product?

mpdehaan commented:
Looking at this now though, just in case I misread that diff :)

mpdehaan commented:
Ok I took a look at this -- while in general I like that it makes the background match and such, if viewed on a very wide monitor, the sentences become very long left to right, which can make them harder to read.

As such, I think I know why the limit was there.

This is somewhat related to why newspaper column widths are the way they are.

Thoughts?

sivel commented:
Hrm, maybe we can just extend max-width to something larger than 800px.  Maybe 1200px or something.  I just don't like looking at web pages on my monitor and seeing a lot of unused space.  I feel like my head is constantly looking off to the left because the text doesn't fill the screen.  

Maybe I should first forget about the width, and we can focus on getting an unminified version of the CSS checked in, and have the makefile minify the file.  That should at least make it easier to make additional CSS updates.

I'll work on that this evening or tomorrow.

I've always uses yuicompressor, so I'd probably be more likely to use that.  Do you have any preference there?

BTW, the uncompressed file is in this pull.  I think it was just that I did a move and an add in the same commit.  The files tab shows all 3 modified files.

sivel committed:
Add *.min.css to docsite/.gitignore

sivel committed:
Add staticmin make target for compressing docs related css and js files, currently only theme.css

sivel commented:
I just messed around with the branch and rebased and squashed some things.

The title and original message has been updated to reflect the current state of this pull request.

mpdehaan commented:
I think something like 1200 might be fine.  

I have a huge iMac at home and it was starting to be a little weird left-to-right reading with the default patch.

The other thing I'd like to do is make the background the same color, though haven't dug in to see what that was yet.

Small pages the grey outside the white box looks a little weird, otherwise it's a pretty good theme.

I'm fine with whatever compressor as long as it's available in Fedora+EPEL+Ubuntu 12.04 LTS so most people can build it ok, assuming we don't check in the build product (which we could decide to do).

Thanks!

sivel commented:
Alright, I'm going to look for something that is available via package managers for minifying css.  yuicompressor may not be that, but we will see.

sivel committed:
Use some sed magic as the css minifier

sivel committed:
Properly indent @media blocks

sivel committed:
Get rid of grey background when content doesn't fill the screen

sivel committed:
Increase the content max-width to 1140px from 800px

sivel commented:
I think this is ready to be tested.

Recent changes:
1. Don't use a CSS minifier, just use some sed magic that I found for minifying CSS.
2. Properly indent media blocks in the CSS file
3. Get rid of the greyish background around the content when the content didn't fill the screen
4. Increase the max-width of the content to 1140px.  This is to match a 1440 width monitor (300 for the sidebar)

I could not find a decent CSS minifying app that all distros seemed to have available.  Some searching led me to a somewhat standard series of sed commands for performing the minifying.  I tested with CSS linting, and there was no noticeable difference between the outcome of using sed to using a well known CSS minifier with the exception that the sed commands do not perform optimizations.

sivel commented:
The CSS additions that were made directly to the .html files in f77e744666f9a58e6848be397ba078732855fe6c should probably be moved into theme.css 

For one, the max-width change made here to 1140px may get overridden to a lower 1024px due it being specified in layout.html

I'll work on trying to move those changes in shortly.

sivel committed:
Don't overwrite max-width for wy-nav-content

sivel commented:
The only thing I modified from https://github.com/ansible/ansible/commit/f77e744666f9a58e6848be397ba078732855fe6c was to remove the max-width override that was added, since that is already covered in theme.css

mpdehaan commented:
Checking this out shortly, thanks!

mpdehaan commented:
Merged in, very nice stuff (and clever minify!), will appear next time we push docs.

Thanks!
